src/_pytest/terminal.py

```diff
@@ -457,33 +465,38 @@ def pytest_runtest_logreport(self, report: TestReport) -> None:
                 self._tw.write(" " + line)
                 self.currentfspath = -2
 
+    @property
+    def _is_last_item(self):
+        return len(self._progress_nodeids_reported) == self._testscollected
+
+    # XXX
+    if "xdist" in sys.modules:
+
+        def pytest_xdist_node_collection_finished(self, node, ids):
+            self._testscollected = len(ids)
```

Comment:
This is needed since `_session` is gone (and used by xdist).
I think it's good to remove that "hack", but it needs to only add the `pytest_xdist_node_collection_finished` for when it is recognized.
Can this be improved?